features/admin.users.v1/components/user-groups-list.tsx

```diff
@@ -0,0 +1,493 @@
+        handleUserUpdate,
+        "data-componentid": componentId
+    } = props;
+
+    const { t } = useTranslation();
+
+    const dispatch: Dispatch = useDispatch();
+
+    const [ searchQuery, setSearchQuery ] = useState<string>("");
+    const [ listItemLimit, setListItemLimit ] = useState<number>(UIConstants.DEFAULT_RESOURCE_LIST_ITEM_LIMIT);
+    const [ listOffset, setListOffset ] = useState<number>(0);
+    const [ paginatedGroups, setPaginatedGroups ] = useState<GroupsMemberInterface[]>(undefined);
+    const [ triggerClearQuery, setTriggerClearQuery ] = useState<boolean>(false);
+    const [ showDeleteConfirmationModal, setShowDeleteConfirmationModal ] = useState<boolean>(false);
+    const [ deletingGroup, setDeletingGroup ] = useState<GroupsMemberInterface>(undefined);
+    const [ isDeleting, setDeleting ] = useState<boolean>(false);
+
+    const userGroupsList: GroupsMemberInterface[] = user?.groups;
+
+    /**
+     * Set initial selected roles options
+     */
+    useEffect(() => {
+        setGroupsPage(0, listItemLimit, userGroupsList, searchQuery);
+    }, [ user, searchQuery ]);
+
+    /**
+     * Util method to paginate retrieved groups  list.
+     *
+     * @param offsetValue - pagination offset value.
+     * @param itemLimit - pagination item limit.
+     * @param list - Role list.
+     */
+    const setGroupsPage = (
+        offsetValue: number,
+        itemLimit: number,
+        list: GroupsMemberInterface[],
+        searchQuery?: string
+    ) => {
+        if (!list) {
+            setPaginatedGroups([]);
+
+            return;
+        }
+
+        if (!isEmpty(searchQuery)) {
+            const filteredList: GroupsMemberInterface[] = list.filter((group: GroupsMemberInterface) => {
+                return group.display.toLowerCase().includes(searchQuery.toLowerCase());
+            });
+
+            setPaginatedGroups(filteredList?.slice(offsetValue, itemLimit + offsetValue));
+
+            return;
+        }
+
+        setPaginatedGroups(list?.slice(offsetValue, itemLimit + offsetValue));
+    };
+
+
+    const handlePaginationChange = (event: React.MouseEvent<HTMLAnchorElement>, data: PaginationProps) => {
+        const offsetValue: number = (data.activePage as number - 1) * listItemLimit;
+
+        setListOffset(offsetValue);
+        setGroupsPage(offsetValue, listItemLimit, userGroupsList);
+    };
+
+    const handleItemsPerPageDropdownChange = (event: React.MouseEvent<HTMLAnchorElement>, data: DropdownProps) => {
+        setListItemLimit(data.value as number);
+        setGroupsPage(listOffset, data.value as number, userGroupsList);
+    };
+
+    const handleSearchQueryClear = (): void => {
+        setTriggerClearQuery(!triggerClearQuery);
+        setSearchQuery("");
+    };
+
+    const resolveTotalPages = (): number => {
+        return Math.ceil(userGroupsList?.length / listItemLimit);
+    };
+
+    const unassignUserFromGroup = (group: GroupsMemberInterface) => {
```

Comment:
Let's add doc comments for each function

Reply:
Fixed with cb0ee2cf165319ca096aa44d358799060f08bbf9